### PR TITLE
improve filebeat scan performance when symlinks is enabled

### PR DESCRIPTION
**Pull request by liuwenping**
## What does this PR do?

Improve filebeat file scanning performance when symlinks is enabled

## Why is it important?

On my machine, because it is a log server, the target file will reach tens of thousands or even millions. 
After filebeat is started, the CPU has been very high. 
Through golang pprof, most of the time is consumed in the scan->getFiles->os.SameFile function. 
(Symlinks is enabled in my configuration file)
I looked at the code carefully and found that there is an O(n2) performance loss when traversing `paths`.
On my machine, there are millions of files, and it takes more than 1 hour to run once here

#28328

### CODE REVIEW

File: filebeat/input/filestream/fswatch.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ import (
 	"github.com/elastic/beats/v7/filebeat/input/file"
 	loginp "github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile"
 	"github.com/elastic/beats/v7/libbeat/common"
+	filelibbeat "github.com/elastic/beats/v7/libbeat/common/file"
```
Comment (kvch) [inline filebeat/input/filestream/fswatch.go:29]: Could you please rename the alias to `file_helper`?

**Comment by kvch** [comment]
Nice catch! Once you rename the import alias, I will approve it.

File: filebeat/input/filestream/fswatch.go (hunk 1 above)
Comment (liuwenping) [reply filebeat/input/filestream/fswatch.go]: Thanks for reading my PR, I will fix it right away

**Review by kvch — APPROVED** [review]

**Comment by kvch** [comment]
Thank you!

**Pull request merged**